src/utils/order.js

```diff
@@ -0,0 +1,75 @@
+var debug = require('./debug');
+var warn = debug('utils:order:warn');
+
+/**
+ * Derives an ordering from the elements with before and after constraints.
```

Comment:
order no descriptive. not sure it needs to be a separate utils file. we can just include this function in a-scene

Reply:
The `a-scene.js` file is already pretty lengthy. I'd rather not append this to it, especially given that the implementation details aren't relevant from the POV of the scene.

Definitely open for (file) name suggestions. I left out `component` as it can also be applied to systems down the line, but if it helps clarity now, it can be included in the name.

Reply:
900 LOC is not super big and purpose of `utils` is factor out code used from multiple files or exported. When understanding a code base having tons of files that don't know how to relate to one another a bigger problem than big files. I would just incorporate this code to `a-scene`

Reply:
Don't really agree, but ultimately a subjective point, so added it to `a-scene.js`.